### DISCUSSION TITLE
Q: How to backtesting Micro-Futures

### DISCUSSION THREAD

**Discussion by KingKnecht** [kernc/backtesting.py#314] [Q&A]
When trading Micro-Futures
1. there is a fixed commission say 0.77€/contract. As I understand it's not possible to have fixed commission with this framework?
2. one tick (0.25) in price  is worth 1.25€. Is there a way to simulate this?

Thx

**Comment by kernc** [comment]
> 1. there is a fixed commission say 0.77€/contract. As I understand it's not possible to have fixed commission with this framework?

Indeed, duplicate of https://github.com/kernc/backtesting.py/issues/113. Something simple adversarial as this might work:
```py
commission = ceil(0.77 / data.Close.mean() * 1e3) / 1e3
```

> 2. one tick (0.25) in price  is worth 1.25€. Is there a way to simulate this?

I don't quite get it. One tick (1/4) in price is worth 1.25€? So four ticks (4/4) signify a move of 5€? Can you show a sample of your data?

**Comment by KingKnecht** [comment]
> I don't quite get it. One tick (1/4) in price is worth 1.25€? So four ticks (4/4) signify a move of 5€? Can you show a sample of your data?

Exactly. 
![image](https://user-images.githubusercontent.com/4339172/115971368-5491c600-a548-11eb-93b1-2dee9d203abb.png)

See here:
[CME Group](https://www.cmegroup.com/trading/equity-index/us-index/micro-e-mini-sandp-500_contract_specifications.html)

**Comment by KingKnecht** [comment]
Here are same trading examples. Red lined example is 2 ticks ( tick size 0.25) which is then 2.50€.
![image](https://user-images.githubusercontent.com/4339172/115971641-e77f3000-a549-11eb-8373-c69fc01e7297.png)

**Comment by kernc** [comment]
If you get 5€ per every unit of price, seems to me simply multiplying `data` by 5 should do it.

**Reply by nicolaszabatta** [reply]
In fact there are many more factors in futures trading that are different from stocks.

Futures have an initial margin and intraday margin for some brokers. This mean you can open a position controlling a notional value of around a million dollars with $500. So there is no leverage like in the CFD or Forex brokers like 1 to 100, there is a certain amount of money to open a huge position per contract. This is why it's appealing for day trading. 

Also, the minimum price fluctuation is called Tick Size, which is different in every futures contract, and each tick has a Tick Value.
